R/utils-get_code_dependency.R

```diff
@@ -144,6 +145,16 @@ fix_shifted_comments <- function(calls) {
   Filter(nrow, calls)
 }
 
+#' Fixes edge case of `:=` assignment operator being treated as assignment.
+#' @keywords internal
+#' @noRd
+remove_dt_assign <- function(calls) {
+  checkmate::assert_list(calls)
+  lapply(calls, function(call) {
+    call <- subset(call, !(token == "LEFT_ASSIGN" & text == ":="))
```

Comment:
According to documentation `subset` should be avoided, what do you think of the suggestion below?

![image](https://github.com/user-attachments/assets/39b5ff07-bcfb-41be-8c09-4c58972dbbb0)

```suggestion
    call[!(token == "LEFT_ASSIGN" & text == ":="), ]
```